src/bindgen/ir/function.rs

```diff
@@ -64,6 +70,42 @@ impl Function {
             ret.replace_self_with(self_path);
         }
 
+        // Check for associated type in ret
+        let mut ret_assoc = None;
+        if let syn::ReturnType::Type(_, ret_type) = &sig.output{
+            if let syn::Type::Path(type_path) = ret_type.as_ref(){
+                if let Some(_) = type_path.qself{ 
+                    let assoc_types_key = AssocTypesKey::new(&type_path);
+                    if assoc_types_key.is_err(){
+                        warn!("Qualified type found in function return type but unable to parse trait.");
+                    }
+                    else{
+                        ret_assoc = Some(assoc_types_key.unwrap());
+                    }
+                }
+            }
+        };
+
+        // Check for asociated types in arg
+        let mut arg_assoc: Vec<Option<AssocTypesKey>> = Vec::new();
+        for arg in sig.inputs.iter(){
```

Comment:
```suggestion
        for arg in &sig.inputs{
```

I believe `iter` is implicit